worker/threatgrid/threatgrid/threatgrid.py

```diff
@@ -0,0 +1,131 @@
+#   Copyright 2014-2016 PUNCH Cyber Analytics Group
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""
+Overview
+========
+
+Interact with ThreatGrid API
+
+"""
+
+import argparse
+import pythreatgrid
+import threading
+
+from stoq.args import StoqArgs
+from stoq.plugins import StoqWorkerPlugin
+from time import sleep
+
+
+class ThreatGridWorker(StoqWorkerPlugin):
+
+    def __init__(self):
+        super().__init__()
+        self.threadgrid_lock = threading.Lock()
+        self.wants_heartbeat = True
+
+    def activate(self, stoq):
+        self.stoq = stoq
+
+        parser = argparse.ArgumentParser()
+        parser = StoqArgs(parser)
+
+        worker_opts = parser.add_argument_group("Plugin Options")
+        worker_opts.add_argument("--api_host",
+                                 dest='api_host',
+                                 help="ThreatGrid API host")
+        worker_opts.add_argument("--vm",
+                                 dest='vm',
+                                 default=None,
+                                 help="Specified virtual machine to use")
+        worker_opts.add_argument("--private",
+                                 dest='private',
+                                 default='false',
+                                 help="Specify if sample should be marked private or not")
+        worker_opts.add_argument("--api_key",
+                                 dest='api_key',
+                                 help="ThreatGrid API Key")
+        worker_opts.add_argument("--tags",
+                                 dest='tags',
+                                 default=[],
+                                 help="Tags applied to samples")
+        worker_opts.add_argument("--playbook",
+                                 dest='tags',
+                                 default='none',
+                                 help="Name of playbook to apply to sample run")
+        worker_opts.add_argument("--network_exit",
+                                 dest='network_exit',
+                                 default='phl-ven',
+                                 help="ThreatGrid network exit to use")
+
+        options = parser.parse_args(self.stoq.argv[2:])
+        super().activate(options=options)
+
+        return True
+
+    def deactivate(self):
```

Comment:
This is probably something that should be put in the framework, rather than here as part of the cleanup process.

Reply:
Good catch. This was old cruft emulating one of the other older plugins.

Reply:
Do you also want the "return True" nuked as well?